src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -234,6 +330,55 @@ BuildSelectStatement(Query *masterQuery, List *masterTargetList, CustomScan *rem
 
 
 /*
+ * make_unique_from_sortclauses creates and returns a unique node
+ * from provided distinct clause list.
+ * The functions is copied from postgresql function
+ * with the same name from optimizer/plan/createplan.c.
+ */
+static Unique *
+make_unique_from_sortclauses(Plan *lefttree, List *distinctList)
```

Comment:
Thinking about this a bit more, I think, we should move this to separate file similar to what we did for ruleutils.c.  Source code also says that external use of `make_sort_from_sortclauses` is deprecated, so wherever they move it from` planmain.h`, we can also put it to the same file. What do you think?

Reply:
I agree, we should have some placeholder for such copy/paste functions. Do you know our existing approach here ? ```ruleutils``` is very specific, we should have more general file perhaps. There is a risk that we may have to differentiate on version numbers.

@marcocitus  may have an idea here.

Reply:
moving the function into a new file  ```postgres_planning_functions.c```